### PR TITLE
Adds global configuration file

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

zonemaster-cli has support for a configuration file, cli.args, for each user, with the same arguments as on the command line. #252 requests a global configuration file. This PR adds a possibility to have such a file.

The global configuration file is hard coded to be /etc/zonemaster/cli.args (Linux style) or /usr/local/etc/zonemaster/cli.args (FreeBSD style), i.e. the same directory as the Backend configuration is stored in by default.

Arguments on the command line takes precedence over the equivalent argument in ~/.zonemaster/cli.args, which takes precedence over the equivalent argument in a global cli.args. Only one global file is used, /etc/zonemaster/cli.args if that exists, else /usr/local/etc/zonemaster/cli.args.

The way the cli.args is read has been updated to support empty lines and lines starting with "#", comments.

## How to test this PR

Create ~/.zonemaster/cli.args, /etc/zonemaster/cli.args and /usr/local/etc/zonemaster/cli.args and verify that arguments are used and the the precedence order works.

### CODE REVIEW

**Comment by matsduf** [comment]
@aabdnn, this is a proposal to give global configuration file (cli.args) for zonemaster-cli.

File: script/zonemaster-cli [hunk 1]
```diff
@@ -7,16 +7,51 @@ use Zonemaster::CLI;
...
-    chomp(@lines);
+if ( defined $globalconffile ) {
+    open my $fh, '<', $globalconffile;
+    my @lines;
+    while (<$fh>) {
+        chomp;
+        next if /^\s*$/;
+        next if /^\s*#/;
+        push @lines, $_;
+    };
```
Comment (pnax) [inline script/zonemaster-cli:50]: Did you consider making a function to avoid code duplication when reading each file?

Comment (matsduf) [reply script/zonemaster-cli]: Fixed.

**Comment by matsduf** [comment]
@mattias-p and @pnax, please re-review.

**Comment by mattias-p** [comment]
I came to think about something. Global configuration of command line parameters could be achieved with zero lines of code in zonemaster-cli by adding an alias to /etc/profile. I suggest this PR is updated to simply explain how this is done in the "Configuration" POD section.

A possible further step in this direction would be to deprecate the cli.args feature and instruct users to replace their cli.args file with an alias in $HOME/.profile.

When we finally drop the cli.args feature:
* We'll have a uniform format for both system-wide and per-user configuration of command line parameters.
* We'll impose one one fewer configuration formats on the world. (The use of alias is already established.)
* We achieve this in a negative number of lines of code. Meaning less room for bugs.

Thoughts on this?

**Comment by matsduf** [comment]
> I came to think about something. Global configuration of command line parameters could be achieved with zero lines of code in zonemaster-cli by adding an alias to /etc/profile. I suggest this PR is updated to simply explain how this is done in the "Configuration" POD section.

A global alias in /etc/profile works for those that use SHELL that uses `/etc/profile`. It could require that aliases are created in different files for different SHELL.

> A possible further step in this direction would be to deprecate the cli.args feature and instruct users to replace their cli.args file with an alias in $HOME/.profile.

Not all SHELL uses .profile so the instruction would be more complex.

> Thoughts on this?

I find the solution with `cli.args` to be more robust and user friendly. The code needed is not complex and it is easy to test. There is a clear use cases for both private and global `cli.args`.

**Comment by mattias-p** [comment]
> A global alias in /etc/profile works for those that use SHELL that uses /etc/profile. It could require that aliases are created in different files for different SHELL.

> Not all SHELL uses .profile so the instruction would be more complex.

All shells in the Bourne shell family use /etc/profile and .profile.

Notably the Csh family does not use these particular files. Though surely anyone who uses one of those shells is an advanced user how has no trouble configuring their system-wide and per-user aliases even when the documentation only refers /etc/profile and .profile.

**Comment by matsduf** [comment]
Csh is the default shell for root user in FreeBSD.

It is also simpler to handle a cli.arg file since changes take effect directly without having any special reload. Also update of cli.args does not effect anything else.

**Comment by matsduf** [comment]
@pnax and  @mattias-p, please re-review.

**Review by pnax — APPROVED** [review]
lgtm

**Review by mattias-p — APPROVED** [review]

**Comment by matsduf** [comment]
@aabdnn, it will be included in the upcoming release.

**Pull request merged**